lxd/provider.go

```diff
@@ -367,3 +348,66 @@ func validateLxdRemoteScheme(v interface{}, k string) ([]string, []error) {
 	}
 	return nil, nil
 }
+
+// InitClient creates and returns an LXD client for the named remote
+func (p *LxdProvider) initClient(remote string) (lxd.Server, error) {
+	var client lxd.Server
+	var err error
+
+	if p.Config.Remotes[remote].Protocol == "simplestreams" {
+		client, err = p.Config.GetImageServer(remote)
+	} else {
+		client, err = p.Config.GetContainerServer(remote)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if p.clientMap == nil {
+		p.clientMap = make(map[string]lxd.Server)
+	}
+
+	p.clientMap[remote] = client
+	return client, nil
+}
+
+// GetClient returns an LXD client for the named remote
+func (p *LxdProvider) GetClient(remote string) (lxd.ContainerServer, error) {
```

Comment:
The naming of this and `GetServerClient()` below are kind of confusing and backwards - I should improve.